### DISCUSSION TITLE
What goes where

### DISCUSSION THREAD

**Discussion by JimMadge** [alan-turing-institute/REG-handbook#8] [Ideas]
Some of the knowledge we have to share or articles we would like to write could be useful far beyond REG.

We should aim to not duplicate effort or have an overlap of purpose with other guides. Our impact will be greater if we target books with a large audience for widely applicable topic. In particular, we should aim to support [The Turing Way](https://github.com/alan-turing-institute/the-turing-way).

The [Netherlands eScience center guide](https://guide.esciencecenter.nl/) uses a system where team specific items goes into the guide and more general/broad articles are targeted for The Turing Way. Some articles are a combinations of team recommendations/preferences and a link to relevant parts of The Turing Way. See the [licensing](https://guide.esciencecenter.nl/#/best_practices/licensing) page for example.

This discussion is a place to sketch out our guidelines for what kind of content belongs in the REG handbook and what should be placed elsewhere. This should become part of the handbook.

It might also be useful to discuss here (or in another discussion) ways to encourage/ensure contribution to The Turing Way

**Comment by JimMadge** [comment]
My attempt at defining what goes here and what doesn't.

Bottom line,

If it is for our team, it goes in the handbook

What belongs in the REG handbook,

- REG specific internal processes
- REG specific ways of working
- Stub articles linking to (or embellishing) external resources (like the licensing example above)
- Information which we refer to frequently (something like an REG filtered FAQs)

What doesn't belong in the REG handbook

- Everything else (?)

**Reply by martintoreilly** [reply]
Is there really a clear line between something that's "for our team" vs something that might be of interest to others 🙃? What makes something a "REG-specific" way of working? Do we mean our particular choice of combination of X, Y, Z practices, in which case I think others would be interested in what choices we made and why. Do we mean ways of working we've developed ourselves that no-one else is using, in which case I think others would also be interested in what we're doing and why. Or do we only mean the things that just don't make sense outside of the REG or Turing organisational context?

By the way, I think many of our "internal" policies and processes would also be of interest to others. There are regular requests in the RSE leaders channel on the Society RSE slack for people to share what their groups do around cost recovery, people allocation, career paths, hybrid working etc, and I think there would be appetite to curate these together across RSE groups as a wider community resource. I can also see some of these areas being natural topics for inclusion in the Turing Way (e.g. they already have a [remote collaboration section](https://the-turing-way.netlify.app/collaboration/remote-collab.html)).

When we are talking about "what belongs where", what exactly are we talking about? It feels like it's a mix of the following.
1. Where content is **published** - i.e. content would be useful in two places, but can currently only be easily published in one place and not re-used as effectively in other places.
2. **Ownership** of the content - i.e. who controls the content and updates to it rather than where it is published.

What are people's thoughts? How much of the "what belongs where" is about how easily content can be effectively re-used and how much is about who controls the content? Or are there other aspects I've missed?

**Reply by JimMadge** [reply]
> Is there really a clear line between something that's "for our team" vs something that might be of interest to others 🙃?

No I don't think so. That is one reason why making everything public unless there is a strong reason not to. I think I wouldn't make the distinction based on what is interesting, but instead who the primary audience is. I think that I what I really mean by "REG-specific". 

For example, a page about using Kanban boards for managing projects has a broad primary audience. A page about the specifics of how REG should use its own, specific, hut23 project tracking board is primarily aimed at REG, even though others might find it useful or interesting.

My idea is that the handbook should be the single, canonical source of REGs specific processes and ways of working. That way we maximise the benefit for us. Ensuring it is public and CC licensed goes some way to making it useful for others but I think it is clear we can do a lot better than that by working closely with The Turing Way.

**Reply by JimMadge** [reply]
> When we are talking about "what belongs where", what exactly are we talking about? It feels like it's a mix of the following.
>
> 1.    Where content is published - i.e. content would be useful in two places, but can currently only be easily published in one place and not re-used as effectively in other places.
> 2.    Ownership of the content - i.e. who controls the content and updates to it rather than where it is published.

In practice these feel like the same thing. The admins/maintainers of the repository will have editorial control and (most likely) control over the deployment.

Legally, the owner is the copyright holder with terms defined by the licence. In this case CC means you can always fork, modify, distribute. However, I don't think we want to rely on "you can always fork it".

**Comment by JimMadge** [comment]
@rolyp @evelinag @crangelsmith @malvikasharan @kirstiejane I think you may have opinions on this to share.

@malvikasharan It would be great if you could write something about how REG specific articles could be case studies in The Turing Way and the value of case studies for new RSE groups.

**Comment by malvikasharan** [comment]
Thank you for tagging us Jim.
As we were discussing, knowledge from the Turing's RSE approaches will be valuable for other orgs that are setting their RSE teams within the UK, and more significantly in countries where RSE is a new concept.

We support the development of case study chapter by chapter (as you can see in the Guide for Collaboration: https://the-turing-way.netlify.app/collaboration/research-infrastructure-roles/data-steward/data-steward-casestudy.html?highlight=case%20study) or for the entire guide (as in the guide for reproducibility: https://the-turing-way.netlify.app/reproducible-research/case-studies.html).
Here is a template with guidance on what a case study might look like: https://github.com/alan-turing-institute/the-turing-way/tree/main/book/templates/case-study-template.

As we finalise on our UX/UI design for providing different entry points for different user groups, we should be able to curate existing chapters in The Turing Way for the RSE members, or add more chapters in that curated view that include case studies from the Turing REG.

**Reply by martintoreilly** [reply]
There are regular conversations on the Society RSE Slack channels about various aspects of research software engineering good practice. I think it would be interesting to capture the choices different groups have made and identify where there is consensus on the best approach and where different groups make different choices and why.

**Comment by JimMadge** [comment]
Relevant extracts from an email thread,

@JimMadge 

> I spoke with Malvika Sharan and Carlos Martinez (from Netherlands eScience Center) today about the REG handbook and how it might interact with The Turing Way.
>
> Carlos is the community manager for the Netherlands eScience Center’s [guide](https://guide.esciencecenter.nl/#/). The guide aims to upstream articles to The Turing Way when the audience is broader than just their team. I think this is something we should aim for in the handbook. Carlos explained the importance of the community manager role to help encourage contributions, organise book dashes and keep the contents up to date.
>
> Malvika Sharan has recently hired a number of community managers for The Turing Way and has opened the possibility of a community manager working with us (REG) on the handbook and The Turing Way.
>
> I will let Malvika Sharan explain any details so that I don’t misrepresent her or the role.

@martintoreilly 

> I’d really love us to follow the Netherlands eScience Center model and have as much as possible of our operating guide be new / extended / updated content in the Turing Way. It would be great to have one of Malvika’s team of community managers supporting us in this!
>
> I’d like to think we could have even more of our content part of the Turing Way than the Netherlands eScience Center. Could we potentially have the REG handbook be essentially just one curated set of topics using the new UX functionality we’re working on with TPS?

@malvikasharan 

> Thanks Jim, for summarising today’s meeting.
 
> I agree with Martin, that we can take advantage of what May, Lydia and Iain are working on within The Turing Way to provide specific version/view of the book based on the user profile as shown in this [test version](https://the-turing-way-choose-your-own-adventure.netlify.app/pathway/dsg/welcome.html) (read [details here](https://github.com/alan-turing-institute/the-turing-way/issues/2071)). Among a few user profiles, we can set one for RSE or more specifically for Research Engineering Group (REG) at the Turing, so that folks in REG can work directly in The Turing Way and don’t have to worry about maintaining two set of documents.
>
> Though you have already read about them, I would like to formally introduce the new TPS members, Arron, Achintya and Vicky who are community managers on different projects (read [more here](https://www.turing.ac.uk/research/research-programmes/tools-practices-and-systems/people)). I see that Arron’s profile has not been updated, but he is the person I would like to specifically connect to support Jim and Roly in this project so that we can ensure some support in getting REG process documented in The Turing Way. But of course, all of us can be reached via an email, GitHub issue on The Turing Way repo or the #tps Slack channel.
>
> I want to emphasise some points from our meeting with Carlos: he has been collaborating with The Turing Way over 2 years and has faced some challenges that REG members face. This include identifying incentives for documenting work as RSE members jump from project to project, building accountable spaces such as Book Dash where people can come together and dedicated time as a team to reflect on their practices (what works and what didn’t work), maintaining GitHub based track record during an ongoing project to share support needed in the project and so on. He is a great person to connect with as we identify challenges in this project and more specifically assess how we engage with the REG members.
>
> I think a few members from within REG to involve in this project, who are already working with us are Camila, Callum, Luke, Lydia, Iain, Fede and anyone who loved open source and is looking for more ways to stay involved.

@JimMadge 

> I’m not convinced that a handbook as a slice of The Turing Way is the best option.
 >
> I think there are strong advantages to REG keeping sole ownership and editorial control of its handbook. We can focus on keeping the processes fast and tailored for our convenience. Fewer stakeholders and less content will simplify designing, managing and deploying the book. Contributing strictly REG content to The Turing Way and tagging it as REG so we can pull it out later seems like an overengineered way to achieve the same as a more KISS approach would. The problem of deciding what goes where remains in any case.
 >
> My main concern is making something that works for REG, I’m worried that going that way would make using and contributing to the handbook more difficult. We already expect ensuring contributions are made and articles are kept up to date to be big challenges. We need a critical mass of users/contributors for the handbook to become part of the team’s culture, so I want to avoid as many barriers as possible.

@martintoreilly 

> I appreciate the desire to keep the barrier to contribution low to ensure that contributing to and maintaining the handbook becomes a core part of the team culture. However, I’d also like working in the open with others on things where we share a common interest to be a core part of the team culture 😁, and the Turing Way is exceptionally well set up for supporting this.
 >
> I may well have been too ambitious in suggesting we make the REG handbook purely a curated view of the Turing Way. However, I am keen for us to push as much as possible of our content into the Turing Way, which is the flagship way we as an Institute are sharing advice and guidance on how to do data science well with the world. It feels like content about how we do data science and research software development at the Turing would be a valuable part of that.
 >
> In terms of connecting with the Turing Way / Jupyter Book UX work, I think the main thing I’d like to see is us making the reuse and remixing of content across different contexts easier, and it feels like the contribution to and reuse of Turing Way content as part of the REG handbook could perhaps be an interesting concrete use case to inform this work.
 > 
> Shall we merge this conversation into the “What goes where” discussion on the handbook repo? I see there’s already a discussion started there about the possibility of using case studies within the Turing Way to capture some of the particular choices REG make around various aspects of our ways of working.

@malvikasharan 

> As Martin said, through The Turing Way, we want to share recommendations, examples and practices on how to do data science well drawing from our work at the Turing. I see it as our responsibility as researchers at the national centre for data science.
> 
> TPS doesn’t have any separate policy from the Turing institute (Kirstie, please correct me here). As a research programme, we want to ensure transparent reporting, interoperability, ethical integrity and respectful co-creation ([details](https://www.turing.ac.uk/research/research-programmes/tools-practices-and-systems)). Our open ways of working don’t mean opening up everything but keeping “as open as possible, as closed as necessary”. Furthermore, we do not intend to add extra burden on anyone – but collaborate to integrate open and reproducible practices in their work, a part of which is to maintain documentation and communicate them openly.
>
> As said, I think everything you consider important internally can be useful for anyone trying to understand RSE roles, become an RSE or set an RSE team beyond the Turing. If we can approach your work with that understanding, I can imagine there is very little (beyond names and exact specifics of projects) that can’t be shared in The Turing Way.
>
> I would let you have this discussion internally to assess what motivates and incentivises REG members in communication and reporting their work through The Turing Way or otherwise. I am happy to come along and work together on a model that fits well with the current ideas – can you please think about setting a 1-2 hour brainstorming, planning meeting?
 >
> Once we have some sort of shared understanding, we can make bring Arron on board to provide community management support (he is an RSE himself and hence a great person to work on this project).
 >
> Also, yes please, let’s have this conversation openly so we move away from assumptions and have more people add their thoughts.
> Thank you again for thinking about the processes and issues in detail.

**Reply by malvikasharan** [reply]
Thank you for putting this thread here @JimMadge. Since this conversation points to planning a brainstoriming/strategy session, whould we create an issue on that and block 1-2 hour in our calendar for January?

**Reply by JimMadge** [reply]
Hi @malvikasharan,

Sorry for the time it has taken to respond (holiday followed by catching up after holiday!).

We (REG) used one of our regular meetings to talk about the handbook. There was a strong consensus for keeping an independent REG handbook for our highly opinionated or team specific information (team structure, regular events calendar, licensing preference, internal document templates and so on).

The team were also enthusiastic about using the handbook as a case study in The Turing Way and contributing less team-specific or more broadly interesting material whenever it makes sense.

That still leaves questions about how to make that relationship as the best it can be and what processes/people we might want to put in place to make collaborating smooth. I'm still keen to use this an an opportunity to promote contribution to the The Turing Way. It is still very early days for the handbook, do you think it is work having those conversations now?

**Comment by malvikasharan** [comment]
Hi @JimMadge, thank you for sharing the feedback. I would love to  have a collaborative meeting for brainstorming and coworking on defining some of these decision-making processes for when something ends up in REG book and when they should be in TTW. I would like to invite Carlos (eScience Netherlands) to the meeting to help us create a design that can be reused by other teams beyond REG who want to adopt this hybrid model. Can you check if that would be possible?
